File: virtual-patient-api/app/agents/schemas/clinical_case.py
"""
Schema for Clinical Case translation
"""
from typing import Optional, Dict
from pydantic import BaseModel


class ClinicalCaseTranslatableFields(BaseModel):
    """Schema for translatable fields of a clinical case"""
    title: Optional[str] = None
    description: Optional[str] = None
    chief_complaint: Optional[str] = None
    present_illness: Optional[str] = None
    personal_medical_history: Optional[str] = None
    surgical_history: Optional[str] = None
    family_history: Optional[str] = None
    medications: Optional[str] = None
    habits: Optional[str] = None
    allergies: Optional[str] = None
    concerns: Optional[str] = None
    physical_requirements: Optional[str] = None
    socioeconomic_status: Optional[str] = None
    patient_context: Optional[str] = None
    female_name: Optional[str] = None
    male_name: Optional[str] = None

File: virtual-patient-ui/src/types/clinicalCase.ts
export type CaseType = 'default' | 'custom';

export type CreatedByUser = {
  id: number;
  firstName: string;
  lastName: string;
};

export type ClinicalCaseBase = {
  id: number;
  title: string;
  description: string;
  caseType: CaseType;
  organizationId: number | null;
  active?: boolean;
  createdBy?: CreatedByUser | null; // Only for custom cases
  icon?: string | null;
  age: number | null;
  weightInKg?: number | null;
  femalePhoto?: string | null;
  malePhoto?: string | null;
  femaleName?: string | null;
  maleName?: string | null;
  genderRestriction?: string | null;
};


export type PatientFields = {
  weightInKg: number;
  physicalRequirements: string;
  socioeconomicStatus: string;
  patientContext: string;
  chiefComplaint: string;
  presentIllness: string;
  personalMedicalHistory: string;
  surgicalHistory: string;
  familyHistory: string;
  medications: string;
  habits: string;
  allergies: string;
  concerns: string;
};

export type TranslatedClinicalCase = {
  titleTranslations: {
    es: string;
  };
  descriptionTranslations: {
    es: string;
  };
  chiefComplaintTranslations: {
    es: string;
  };
  presentIllnessTranslations: {
    es: string;
  };
  personalMedicalHistoryTranslations: {
    es: string;
  };
  surgicalHistoryTranslations: {
    es: string;
  };
  familyHistoryTranslations: {
    es: string;
  };
  medicationsTranslations: {
    es: string;
  };
  habitsTranslations: {
    es: string;
  };
  allergiesTranslations: {
    es: string;
  };
  concernsTranslations: {
    es: string;
  };
  physicalRequirementsTranslations: {
    es: string;
  };
  socioeconomicStatusTranslations: {
    es: string;
  };
  patientContextTranslations: {
    es: string;
  };
};

export type ClinicalCase = ClinicalCaseBase &
  PatientFields & {
    createdAt: string;
    updatedAt: string;
    organization: any | null;
  };

  export type ClinicalCaseWithTranslations = ClinicalCaseBase &
  PatientFields & TranslatedClinicalCase & {
    createdAt: string;
    updatedAt: string;
    organization: any | null;
  };